### PR TITLE
Add cmake controls for repo tags

### PR DESCRIPTION
**Pull request by deadeyegoodwin**

### CODE REVIEW

**Review by GuanLuo — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -325,6 +325,10 @@ COPY --from=tritonserver_onnx /workspace/build/Release/testdata/custom_op_librar
 # - Need to find CUDA stubs if they are available since some backends
 # may need to link against them. This is identical to the logic in TF
 # container nvbuild.sh
+ARG TRITON_COMMON_REPO_TAG=main
+ARG TRITON_CORE_REPO_TAG=main
+ARG TRITON_BACKEND_REPO_TAG=main
```
Comment (CoderHam) [inline Dockerfile:330]: We would need to have these build args be overridden by the gitlab CI (to be the branch name?). Moreover we would want to set/overwrite these values using CI variables.

Comment (deadeyegoodwin) [reply Dockerfile:330]: Typically you just want to use the main branch of the dependent repos. If your CI run requires a change form one of the repos then you edit Dockerfile and include that in CI run. And then need to remove it before merging. I agree that is awkward and we will have something better once the reorg is further along.

Comment (deadeyegoodwin) [reply Dockerfile:330]: But we could enhance CI to have addition vars that could be passed into the build

Comment (CoderHam) [reply Dockerfile:330]: I think enhancing the CI route would be cleaner. Nevertheless these changes look good and can be merged in

**Review by CoderHam — APPROVED** [review]

**Pull request merged**